src/software/ai/navigator/path_planner/bang_bang_trajectory_1d.cpp

```diff
@@ -0,0 +1,310 @@
+#include "software/ai/navigator/path_planner/bang_bang_trajectory_1d.h"
+
+#include <cmath>
+
+#include "software/geom/algorithms/is_in_range.h"
+#include "software/logger/logger.h"
+
+BangBangTrajectory1D::BangBangTrajectory1D()
+{
+    // Add a default trajectory part to avoid segfaults
+    // when the getters are called before generate
+    addTrajectoryPart(TrajectoryPart());
+}
+
+void BangBangTrajectory1D::generate(double initial_pos, double final_pos,
+                                    double initial_vel, double max_vel, double max_accel,
+                                    double max_decel)
+{
+    // We will overwrite the previous trajectory
+    num_trajectory_parts = 0;
+
+    CHECK(max_vel != 0) << "Max velocity cannot be 0";
+    CHECK(max_accel != 0) << "Max acceleration cannot be 0";
+    CHECK(max_decel != 0) << "Max deceleration cannot be 0";
+
+    // Unify the signs of the kinematic constraints
+    max_accel = std::abs(max_accel);
+    max_decel = std::abs(max_decel);
+    max_vel   = std::abs(max_vel);
+
+    // From initial position, where the closest position is where we can stop.
+    // If it is not between initial and final position, then we must break right away.
+    double stop_pos = closestPositionToStop(initial_pos, initial_vel, max_decel);
```

Comment:
Which paper are you referring to? I had to make some adjustments to the initial deceleration portion since we support different `max_accel` and `max_decel`, which makes the logic of stopping, and accelerating in a different direction a bit less trivial than the other implementations.

Reply:
I was referring to the cases in the paper you pointed to: "Trajectory generation for four wheeled omnidirectional vehicles" which had the different cases but it seems that your implementation might be sufficiently different from the original paper